apps/ltn/src/save/mod.rs

```diff
@@ -36,15 +37,33 @@ impl Proposal {
             abst_version: map_gui::tools::version().to_string(),
 
             partitioning: app.session.partitioning.clone(),
-            modal_filters: app.session.modal_filters.clone(),
+            edits: app.session.edits.clone(),
         }
     }
 
     fn make_active(self, ctx: &EventCtx, app: &mut App) {
+        // First undo any one-way changes
+        let mut edits = app.map.new_edits();
+        for r in app.session.edits.one_ways.keys().cloned() {
+            // Just revert to the original state
+            edits.commands.push(app.map.edit_road_cmd(r, |new| {
+                *new = EditRoad::get_orig_from_osm(app.map.get_r(r), app.map.get_config());
+            }));
+        }
+
         app.session.proposal_name = Some(self.name);
         app.session.partitioning = self.partitioning;
-        app.session.modal_filters = self.modal_filters;
-        app.session.draw_all_filters = app.session.modal_filters.draw(ctx, &app.map);
+        app.session.edits = self.edits;
+        app.session.draw_all_filters = app.session.edits.draw(ctx, &app.map);
+
+        // Then append any new one-way changes. Edits are applied in order, so the net effect
+        // should be correct.
+        for (r, r_edit) in &app.session.edits.one_ways {
```

Comment:
... but I kind of bet that we'll slowly leak memory as people fiddle with undo, switching proposals, and making edits! Probably need to throw in some `edits.compress()` somewhere.